styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/LaunchPlanIdentifier.java

```diff
@@ -0,0 +1,54 @@
+/*-
+ * -\-\-
+ * Spotify Styx Testing Utilities
+ * --
+ * Copyright (C) 2016 - 2019 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.flyte.client;
+
+import com.google.auto.value.AutoValue;
+import flyteidl.core.IdentifierOuterClass;
+
+@AutoValue
+public abstract class LaunchPlanIdentifier {
```

Comment:
Why do we need this class?

```
WorkflowExecutionIdentifier createExecution(String domain, String project,
                                              LaunchPlanIdentifier launchPlanId,
                                              ExecutionOuterClass.ExecutionMetadata.ExecutionMode executionMode)
```
Here we accept both the one generate from proto as well as our own. This looks weird.

Do we have other usage of this class? We might prefer not to remap proto models when the remapped class is not used for anything else.

Reply:
It's definitely not used yet, but should be used when we start utilising the client.
I agree that using the Protobuf as an argument here makes it confusing, I can create an enum that maps to the protobuf instead.

Reply:
What do you think?

Reply:
pushed what I described in https://github.com/spotify/styx/pull/831/commits/1426b02c7e2b0a2a74ba41acde908786c48d80ad

Reply:
I don't it is needed to create a remap. Why not using classes generated from proto? What is the benefit having this additional layer?

Reply:
It is considered a good practise in java to keep the generated protobuf code usage as thin as possible and map it to internal models.
The issue is that sometimes there are breaking changes in the way the generated code looks, or there are incompatibilities between versions.
By mapping immediately we make sure that we will only need to refactor a small part of the code when issues like these come up.
Additionally protobuf enforces certain behaviour, like no null support, which might cause confusion later in the codebase.

Reply:
I can understand the reasoning here, but it still sounds like a premature optimization to me. This is already client code that is the last mile towards server, and providing yet another layer of model seems to add unnecessary complexity. If we need models to be used internally in Styx, they should go to styx-common instead of this module. This is the same as kubernetes client we are using, where we just use whatever the lib provides without creating yet another layer.

I suggest we pull in @narape to align on this because he is working on the model I think.

Reply:
hm, I could remove the intermediate objects for now, and expose use only grpc/protobuf,
and we can refactor them in when we see fit. what do you think?

Reply:
SGTM

Reply:
fixed